### PR TITLE
Sample showing how Marten handles Event type namespace or name migration

### PR DESCRIPTION
**Pull request by oskardudycz**
Added sample and documentation showing how event type namespace or name migration can be handled by Marten. 

More advanced scenarios or guidelines will come in follow up PRs.

### CODE REVIEW

File: documentation/documentation/events/versioning/index.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Marten by default tries to find the event class based on the fully qualified assembly name (it's stored in `mt_dotnet_type` column of `mt_events` table, read more in <[linkto:documentation/events/schema;title=events schema documentation]>). 
+When it is not able to find event type with the same assembly, namespace and type name then it tries to make a lookup for mapping on the event type name (stored in `type` column of `mt_events` table).
+
+Such mapping needs to be defined manually:
+
+- either by registering events with store options `Events.AddEventTypes` method,
+- or by defining custom mapping with `Events.EventMappingFor` method.
+
+For the case of namespace migration, it's enough to use `AddEventTypes` method as it's generating mapping based on the event type. So eg. having `OrderStatusChanged` event changed from:
```
Comment (mysticmind) [inline documentation/documentation/events/versioning/index.md:25]: Request for a small tweak to this sentence, change "So eg. having..." to "As an example, change `OrderStatusChanged` from:"

File: documentation/documentation/events/versioning/index.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+to:
+
+<[sample:new_event_namespace]> 
+
+It's enough to register new event type as follows:
+
+<[sample:event_namespace_migration_options]>
+
+After that Marten will automatically perform a matching based on the type name (that didn't change) - so. `order_status_changed`.
```
Comment (mysticmind) [inline documentation/documentation/events/versioning/index.md:37]: Please remove "so."

File: documentation/documentation/events/versioning/index.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+## Event type name migration
+
+For the case when event class type name changed Marten is unable to perform the automatic mapping. However, it allows to define a custom one.
```
Comment (mysticmind) [inline documentation/documentation/events/versioning/index.md:41]: Request to rephrase this a bit to: "When the event class type name has changed, Marten does not perform automatic mapping but allows to define a custom one."

File: documentation/documentation/events/versioning/index.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+it's needed to register mapping using old event type name (`order_status_changed`) as follows:
+
+<[sample:event_type_name_migration_options]>
+
+<br />
+<div class="alert alert-warning">
+<b><u>Warning:</u></b>
+<br />
+In this case old event type name will be also used when publishing events of the new type name - so both `OrderStatusChanged` and `ConfirmedOrderStatusChanged` will be published with the same `order_status_changed` event type.
```
Comment (mysticmind) [inline documentation/documentation/events/versioning/index.md:57]: Request to change as follows: "In this case both old `OrderStatusChanged` and new `ConfirmedOrderStatusChanged` event type names will get published with the same `order_status_changed` event type."

**Comment by oskardudycz** [comment]
@mysticmind I applied your comments in https://github.com/JasperFx/marten/pull/1523/commits/3212a2d8013ab2a9b2ea3be259914d6caba8c029

**Pull request merged**